### PR TITLE
Support CLUSTER commands

### LINKED ISSUE #1919: Add support for redis command: CLUSTER BUMPEPOCH

### LINKED ISSUE #1920: Add support for redis command: CLUSTER COUNT-FAILURE-REPORTS

### LINKED ISSUE #1921: Add support for redis command: CLUSTER COUNTKEYSINSLOT

### LINKED ISSUE #1922: Add support for redis command: CLUSTER DELSLOTS

### LINKED ISSUE #1924: Add support for redis command: CLUSTER FAILOVER

### LINKED ISSUE #1925: Add support for redis command: CLUSTER FORGET

### LINKED ISSUE #1926: Add support for redis command: CLUSTER KEYSLOT

### LINKED ISSUE #1928: Add support for redis command: CLUSTER MYID

### LINKED ISSUE #1929: Add support for redis command: CLUSTER REPLICAS

### LINKED ISSUE #1930: Add support for redis command: CLUSTER REPLICATE

### LINKED ISSUE #1931: Add support for redis command: CLUSTER SAVECONFIG

### LINKED ISSUE #1932: Add support for redis command: CLUSTER SET-CONFIG-EPOCH

### LINKED ISSUE #1933: Add support for redis command: CLUSTER SLAVES

### PR DESCRIPTION
**Pull request by Avital-Fine**
### Description

Support CLUSTER [...] commands

closes #1933, closes #1932, closes #1931, closes #1933, closes #1930, closes #1929, closes #1928, closes #1926, closes #1925, closes #1924, closes #1922, closes #1921, closes #1920, closes #1922, closes #1919

---

### Checklist

- [x] Does `npm test` pass with this change (including linting)?
- [X] Is the new or changed code fully tested?
- [X] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

### CODE REVIEW

File: packages/client/lib/commands/CLUSTER_INFO.spec.ts [hunk 1]
```diff
@@ -43,4 +44,11 @@ describe('CLUSTER INFO', () => {
             }
         );
     });
+
+    testUtils.testWithCluster('cluster.clusterInfo', async cluster => {
```
Comment (leibale) [inline packages/client/lib/commands/CLUSTER_INFO.spec.ts:48]: Change the title to "clusterNode.clusterInfo" (not just here, in all the tests that uses `getSlotMaster`)

File: packages/client/lib/commands/CLUSTER_SET-CONFIG-EPOCH.spec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        );
+    });
+
+    testUtils.testWithCluster('cluster.clusterSetConfigEpoch', async cluster => {
+        try {
+            assert.equal(
+                await cluster.getSlotMaster(0).client.clusterSetConfigEpoch(1),
+                'OK'
+            );
+        } catch (ReplyError) {
```
Comment (leibale) [inline packages/client/lib/commands/CLUSTER_SET-CONFIG-EPOCH.spec.ts:19]: ?

Comment (Avital-Fine) [reply packages/client/lib/commands/CLUSTER_SET-CONFIG-EPOCH.spec.ts:19]: Didn't know how to handle the error: 
`ReplyError: ERR The user can assign a config epoch only when the node does not know any other node.`

Comment (Avital-Fine) [reply packages/client/lib/commands/CLUSTER_SET-CONFIG-EPOCH.spec.ts:19]: Delete this test?

Comment (leibale) [reply packages/client/lib/commands/CLUSTER_SET-CONFIG-EPOCH.spec.ts:19]: yup

**Pull request merged**